include/swift/AST/Decl.h

```diff
@@ -6770,6 +6763,37 @@ enum class CtorInitializerKind {
   Factory
 };
 
+/// Specifies the kind of initialization call performed within the body
+/// of the constructor, e.g., self.init or super.init.
```

Comment:
Totally optional thought: It could nice to add a sentence like: "Needed to ... when ..."